rust/src/nfs/nfs.rs

```diff
@@ -1859,3 +1928,156 @@ pub extern "C" fn rs_nfs_setfileflags(direction: u8, ptr: *mut NFSState, flags:
+            }
+        } else {
+            AppLayerProtoDetectPPRegister(IPPROTO_TCP as u8,
+                default_port.as_ptr(), ALPROTO_NFS, 0,
+                NFS_MIN_FRAME_LEN, STREAM_TOSERVER,
+                rs_nfs_probe, rs_nfs_probe);
+        }
+        if AppLayerParserConfParserEnabled(
+            ip_proto_str.as_ptr(),
+            parser.name,
+        ) != 0
+        {
+            let _ = AppLayerRegisterParser(&parser, alproto);
+        }
+        SCLogDebug!("Rust nfs parser registered.");
+    } else {
+        SCLogDebug!("Protocol detector and parser disabled for nfs.");
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_nfs_udp_register_parser() {
+    let default_port = CString::new("[2049]").unwrap();
+    let parser = RustParser {
+        name: PARSER_NAME.as_ptr() as *const std::os::raw::c_char,
+        default_port: default_port.as_ptr(),
+        ipproto: IPPROTO_UDP,
+        probe_ts: None,
+        probe_tc: None,
+        min_depth: 0,
+        max_depth: 16,
+        state_new: rs_nfs_state_new,
+        state_free: rs_nfs_state_free,
+        tx_free: rs_nfs_state_tx_free,
+        parse_ts: rs_nfs_parse_request_udp,
+        parse_tc: rs_nfs_parse_response_udp,
+        get_tx_count: rs_nfs_state_get_tx_count,
+        get_tx: rs_nfs_state_get_tx,
+        tx_comp_st_ts: 1,
+        tx_comp_st_tc: 1,
+        tx_get_progress: rs_nfs_tx_get_alstate_progress,
+        get_de_state: rs_nfs_state_get_tx_detect_state,
+        set_de_state: rs_nfs_state_set_tx_detect_state,
+        get_events: Some(rs_nfs_state_get_events),
+        get_eventinfo: Some(rs_nfs_state_get_event_info),
+        get_eventinfo_byid : None,
+        localstorage_new: None,
+        localstorage_free: None,
+        get_files: Some(rs_nfs_getfiles),
+        get_tx_iterator: Some(rs_nfs_state_get_tx_iterator),
+        get_tx_data: rs_nfs_get_tx_data,
+        apply_tx_config: None,
+        flags: APP_LAYER_PARSER_OPT_UNIDIR_TXS,
+        truncate: None,
+    };
+
+    let ip_proto_str = CString::new("udp").unwrap();
+
+    if AppLayerProtoDetectConfProtoDetectionEnabled(
+        ip_proto_str.as_ptr(),
+        parser.name,
+    ) != 0
+    {
+        let alproto = AppLayerRegisterProtocolDetection(&parser, 1);
+        ALPROTO_NFS = alproto;
+
+        if AppLayerProtoDetectPPParseConfPorts(ip_proto_str.as_ptr(), IPPROTO_UDP as u8,
+                parser.name, ALPROTO_NFS, 0, NFS_MIN_FRAME_LEN,
+                rs_nfs_probe_udp_ts, rs_nfs_probe_udp_tc) == 0 {
+            SCLogDebug!("No NFSUDP app-layer configuration, enabling NFSUDP
+                        detection UDP detection on port {:?}.",
+                        default_port);
+            AppLayerProtoDetectPPRegister(IPPROTO_UDP as u8,
+                default_port.as_ptr(), ALPROTO_NFS, 0,
+                NFS_MIN_FRAME_LEN, STREAM_TOSERVER,
+                rs_nfs_probe_udp_ts, rs_nfs_probe_udp_tc);
+        }
```

Comment:
I don't think this bit is required as its already done inside `AppLayerRegisterProtocolDetection`.

Reply:
Oh, but the probe functions aren't set in the registration struct.  If there is a reason for doing this, instead of doing it how it is in `dns.rs` for example, can you elaboroate with a comment as to why?

Reply:
I agree with Jason, `rs_nfs_probe_udp_ts ` should be defined as `probe_ts` in `rs_nfs_register_parser`
Thus, we likely do not need `AppLayerProtoDetectPPRegister` in rust

Reply:
Can we also use `AppLayerProtoDetectPPParseConfPorts` somehow as a part of the registration struct? I used it this way because I did not know how to incorporate conf port check here...